atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClient.java

```diff
@@ -77,20 +107,67 @@ public void batch_mutate(Map<ByteBuffer, Map<String, List<Mutation>>> mutation_m
             super.batch_mutate(mutation_map, consistency_level);
             return null;
         });
+        try {
+            recordBytesWritten(getApproximateWriteByteCount(mutation_map));
+        } catch (Exception e) {
+            log.warn("Encountered an exception when recording write metrics for batch_mutate.", e);
+        }
+    }
+
+    private int getApproximateWriteByteCount(Map<ByteBuffer, Map<String, List<Mutation>>> mutation_map) {
```

Comment:
Might be good to get the Cassandra benchmarks on here as soon as possible, interested to see if streaming through all these objects add any overhead

Reply:
Yeah it may be better to just use regular for loops here, since this is going to be a hot codepath

Reply:
Also, might be nice to refactor out the size calculation logic into a static helper class

Reply:
Refactored to a helper class `ThriftObjectSizeUtils`. We are no longer serializing objects to calculate the size in bytes.